pkg/idl/doc.go

```diff
@@ -133,7 +133,7 @@ type PatchStrategy string
 // used on any struct.
 //
 // Using this tag will generate the following OpenAPI extension:
-//  "x-kubernetes-struct-type": "atomic"
+//  "x-kubernetes-map-type": "atomic"
```

Comment:
Above the comment says: 
> It may have only one value: "atomic".

Clearly that's wrong :-)

Reply:
The entire description might be worth fixing.

Reply:
You should probably also verify if the description for maptype and listtype etc are also correct

Reply:
> Clearly that's wrong :-)
Oh dear 🤦‍♀ fixed it now (hopefully).

Clarified mapType too, as it had a similar problem (I think I got inspired by it to begin with:) ).